tasks/debian.yml

```diff
@@ -1,13 +1,37 @@
 ---
 
-- name: python3 venv | install python3
+- name: install software-properties-common
+  ansible.builtin.apt:
+    name: software-properties-common
+    state: present
+    update_cache: yes
```

Comment:
These decisions have real performance implications for anyone deploying systems using these roles. Unless there is a compelling reason to force `update_cache: {yes, true}`, I would vote for using the default.

Reply:
`apt` on Ubuntu does not have automatic cache expiration. Its metadata remains valid indefinitely unless explicitly updated. However, the Ansible apt module provides a cache_valid_time parameter. I have set `cache_valid_time: 86400` to ensure that the cache is updated if it’s older than one day, offering a balance between performance and freshness.